warcio/limitreader.py

```diff
@@ -7,13 +11,29 @@ class LimitReader(object):
     def __init__(self, stream, limit):
         self.stream = stream
         self.limit = limit
+        self.payload_digester = None
+        self.block_digester = None
+        self.payload_digest = None
+        self.block_digest = None
 
         if hasattr(stream, 'tell'):
             self.tell = self._tell
 
     def _update(self, buff):
         length = len(buff)
         self.limit -= length
+
```

Comment:
Can you split this into a separate subclass, maybe `DigestVerifyingLimitReader()`

The LimitReader is used in a few places, not necessarily to read from WARCs so would be cleaner to just have a separate class that adds the verification.

Reply:
Actually, can probably just be a separate DigestVerifyingReader() that wraps a LimitReader (or any stream).